ProtfolioAndMVVMArchitecture/app/src/main/java/com/mahibul/protfolioandmvvmarchitecture/feacher/home/model/HomeModelImp.kt

```diff
@@ -0,0 +1,30 @@
+package com.mahibul.protfolioandmvvmarchitecture.feacher.home.model
+
+import com.mahibul.protfolioandmvvmarchitecture.core.NetworkCallBack
+import com.mahibul.protfolioandmvvmarchitecture.network.RetrofitApi
+import com.mahibul.protfolioandmvvmarchitecture.network.RetrofitClint
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class HomeModelImp : HomeModel{
+    private val apiInterface = RetrofitClint.clint?.create(RetrofitApi::class.java)
+    private val call = apiInterface?.getUserDetails()
+
+
+    override fun getUserDetails(callBack: NetworkCallBack<UserDetails>) {
+        call?.enqueue(object : Callback<UserDetails> {
+            override fun onResponse(call: Call<UserDetails>, response: Response<UserDetails>) {
+                response.body().let {
+                    if (it != null) {
+                        callBack.onSuccess(it)
+                    }
```

Comment:
should add `else` block. if `it` is null then you should trigger onError() method.